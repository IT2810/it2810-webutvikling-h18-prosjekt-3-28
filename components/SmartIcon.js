import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import {
  Icon
} from 'react-native-elements';

//Had to add this to load font and icons
Expo.Font.loadAsync({
  'Material Icons': require('@expo/vector-icons/fonts/MaterialIcons.ttf'),
  //'FontAwesome': require("expo/vector-icons/FontAwesome")
})

import{
  LinearGradient,
} from 'expo';


class SmartIcon extends React.Component{

  constructor (props) {
    super(props)
    this.state = {
      index: props.index,
      string: props.string,
      parent: props.parent
    }
  }
  
  render () {
    return (
        <TouchableOpacity onPress={() =>this.props.onPress(this.state.index)} activeOpacity={0.5} style={{width: "25%", padding: "2%", borderRadius: "5px", backgroundColor: this.props.backgroundColor}}>
          <Icon name={this.state.string}  color='#00aced' backgroundColor='transparent' size="50" style={styles.icon}></Icon>          
        </TouchableOpacity>
    )
  }
} 

const styles = StyleSheet.create({
  icon: {
    height: "5px",
  }
})

export default SmartIcon