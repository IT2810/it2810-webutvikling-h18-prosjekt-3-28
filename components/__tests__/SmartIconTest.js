import React from 'react';
import SmartIcon from '../SmartIcon';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
      NavigationTestUtils.resetInternalState();
    });
  
    it('renders the loading screen', async () => {
      const tree = renderer.create(<SmartIcon />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the root without loading screen', async () => {
      const tree = renderer.create(<SmartIcon skipLoadingScreen />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

describe('Test render', ()=>{
  const testRenderer = renderer.create(
      <SmartIcon string="work" index={1}></SmartIcon>
    );
})