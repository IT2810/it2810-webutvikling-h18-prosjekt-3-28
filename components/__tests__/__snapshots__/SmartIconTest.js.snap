// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 5,
      "opacity": 1,
      "padding": "2%",
      "width": "25%",
    }
  }
>
  <View>
    <View
      backgroundColor="transparent"
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#323940",
              "fontSize": 50,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 5,
      "opacity": 1,
      "padding": "2%",
      "width": "25%",
    }
  }
>
  <View>
    <View
      backgroundColor="transparent"
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#323940",
              "fontSize": 50,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
